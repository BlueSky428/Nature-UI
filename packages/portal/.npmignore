/src
/node_modules
/test
/.npmrc
/.npmignore
/__tests__