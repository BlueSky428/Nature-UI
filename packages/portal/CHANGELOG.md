# @nature-ui/portal

## 1.0.2

### Patch Changes

- fe90811: Another Realease
- Updated dependencies [fe90811]
  - @nature-ui/hooks@1.0.3
  - @nature-ui/utils@1.0.2

## 1.0.1

### Patch Changes

- 877c868: Major change. Added button-icon component. Included material design
  icons. Included Ion icons icons. Added Left and right icons feature for button
  component.
- Updated dependencies [877c868]
  - @nature-ui/hooks@1.0.2
  - @nature-ui/utils@1.0.1

## 1.0.0

### Major Changes

- 85bb9c8: Major release

### Patch Changes

- Updated dependencies [85bb9c8]
  - @nature-ui/hooks@1.0.0
  - @nature-ui/utils@1.0.0

## 0.2.0

### Minor Changes

- 🐟 Preparing for major release.

### Patch Changes

- Updated dependencies [undefined]
  - @nature-ui/hooks@0.2.0
  - @nature-ui/utils@0.2.0
