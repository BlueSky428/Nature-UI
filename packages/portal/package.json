{
  "name": "@nature-ui/portal",
  "version": "2.0.3-alpha.0",
  "description": "React component used to render children outside the DOM hierarchy of the parent component",
  "keywords": [
    "react",
    "portal",
    "nested portal",
    "z-index",
    "layer",
    "nature ui"
  ],
  "author": "Divine Hycenth <divinehycenth8@gmail.com>",
  "homepage": "https://github.com/DNature/nature-ui",
  "license": "MIT",
  "main": "dist/nature-ui-portal.cjs.js",
  "module": "dist/nature-ui-portal.esm.js",
  "directories": {
    "lib": "dist",
    "test": "__tests__"
  },
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/DNature/nature-ui.git"
  },
  "scripts": {
    "build": "JSX=1 tsup src/index.ts --dts",
    "build:fast": "JSX=1 tsup src/index.ts",
    "dev": "pnpm build:fast -- --watch",
    "clean": "rimraf dist .turbo",
    "typecheck": "tsc --noEmit",
    "prepack": "clean-package"
  },
  "bugs": {
    "url": "https://github.com/DNature/nature-ui/issues"
  },
  "dependencies": {
    "@nature-ui/hooks": "^2.0.3-alpha.0",
    "@nature-ui/react-utils": "^2.0.3-alpha.0",
    "@nature-ui/utils": "^2.0.3-alpha.0"
  },
  "peerDependencies": {
    "react": ">=18",
    "react-dom": ">=16.8.0"
  },
  "devDependencies": {
    "@types/react-frame-component": "^4.1.3",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-frame-component": "^5.2.1"
  }
}
