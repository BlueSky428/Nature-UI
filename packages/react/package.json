{
  "name": "@nature-ui/react",
  "version": "0.1.0",
  "description": "Responsive and accessible React UI components built with react, tailwindcss and emotion",
  "keywords": [
    "react",
    "ui",
    "design-system",
    "react-components",
    "uikit",
    "accessible",
    "components",
    "emotion",
    "library",
    "design-system"
  ],
  "author": "Divine Hycenth <divinehycenth@outlook.com>",
  "homepage": "https://github.com/DNature/nature-ui",
  "license": "MIT",
  "main": "dist/index.js",
  "directories": {
    "lib": "dist",
    "test": "__tests__"
  },
  "files": [
    "dist",
    "src"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/DNature/nature-ui.git"
  },
  "scripts": {
    "build": "rimraf ./dist && tsc",
    "test": "jest --env=jsdom"
  },
  "bugs": {
    "url": "https://github.com/DNature/nature-ui/issues"
  },
  "dependencies": {
    "@nature-ui/alert": "^0.1.0",
    "@nature-ui/alert-dialog": "^0.1.0",
    "@nature-ui/avatar": "^0.1.0",
    "@nature-ui/breadcrumb": "^0.1.0",
    "@nature-ui/button": "^0.1.0",
    "@nature-ui/checkbox": "^0.1.0",
    "@nature-ui/progress": "^0.1.0",
    "@nature-ui/clickable": "^0.1.0",
    "@nature-ui/close-button": "^0.1.0",
    "@nature-ui/collapse": "^0.1.0",
    "@nature-ui/container": "^0.1.0",
    "@nature-ui/drawer": "^0.1.0",
    "@nature-ui/form-control": "^0.1.0",
    "@nature-ui/hooks": "^0.1.0",
    "@nature-ui/icon": "^0.1.0",
    "@nature-ui/image": "^0.1.0",
    "@nature-ui/input": "^0.1.0",
    "@nature-ui/layout": "^0.1.0",
    "@nature-ui/modal": "^0.1.0",
    "@nature-ui/popover": "^0.1.0",
    "@nature-ui/popper": "^0.1.0",
    "@nature-ui/portal": "^0.1.0",
    "@nature-ui/radio": "^0.1.0",
    "@nature-ui/spinner": "^0.1.0",
    "@nature-ui/system": "^0.1.0",
    "@nature-ui/toast": "^0.1.0",
    "@nature-ui/transition": "^0.1.0",
    "@nature-ui/utils": "^0.1.0",
    "@nature-ui/visually-hidden": "^0.1.0"
  },
  "devDependencies": {
    "@nature-ui/system": "^0.1.0",
    "rimraf": "^3.0.2"
  },
  "peerDependencies": {
    "@nature-ui/system": "^0.1.0",
    "react": "16.x",
    "rimraf": "^3.0.2"
  }
}
