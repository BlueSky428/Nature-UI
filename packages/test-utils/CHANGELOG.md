# @nature-ui/test-utils

## 1.1.0

### Minor Changes

- Bump system module

### Patch Changes

- Updated dependencies [undefined]
  - @nature-ui/hooks@1.1.0
  - @nature-ui/system@1.1.0
  - @nature-ui/utils@1.1.0
  - @nature-ui/visually-hidden@1.1.0

## 1.0.2

### Patch Changes

- fe90811: Another Realease
- Updated dependencies [fe90811]
  - @nature-ui/hooks@1.0.3
  - @nature-ui/system@1.0.2
  - @nature-ui/utils@1.0.2
  - @nature-ui/visually-hidden@1.0.2

## 1.0.1

### Patch Changes

- 877c868: Major change. Added button-icon component. Included material design
  icons. Included Ion icons icons. Added Left and right icons feature for button
  component.
- Updated dependencies [877c868]
  - @nature-ui/hooks@1.0.2
  - @nature-ui/system@1.0.1
  - @nature-ui/utils@1.0.1
  - @nature-ui/visually-hidden@1.0.1

## 1.0.0

### Major Changes

- 85bb9c8: Major release

### Patch Changes

- Updated dependencies [85bb9c8]
  - @nature-ui/hooks@1.0.0
  - @nature-ui/system@1.0.0
  - @nature-ui/utils@1.0.0
  - @nature-ui/visually-hidden@1.0.0

## 0.2.0

### Minor Changes

- 🐟 Preparing for major release.

### Patch Changes

- Updated dependencies [undefined]
  - @nature-ui/hooks@0.2.0
  - @nature-ui/system@0.2.0
  - @nature-ui/utils@0.2.0
  - @nature-ui/visually-hidden@0.2.0
