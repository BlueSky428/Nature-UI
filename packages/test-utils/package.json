{
  "name": "@nature-ui/test-utils",
  "version": "0.2.0",
  "description": "Testing utilities for Chakra UI's components",
  "keywords": [
    "nature ui",
    "test utils",
    "testing",
    "utils",
    "testing library",
    "react testing library"
  ],
  "author": "Divine Hycenth <divinehycenth@outlook.com>",
  "homepage": "https://github.com/DNature/nature-ui",
  "license": "MIT",
  "main": "src/index.ts",
  "directories": {
    "lib": "src",
    "test": "__tests__"
  },
  "files": [
    "src"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/DNature/nature-ui.git"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "lint:src": "eslint --fix --ext=ts,tsx src -c ../../.eslintrc",
    "lint:types": "tsc --noEmit",
    "lint": "concurrently yarn:lint:*"
  },
  "bugs": {
    "url": "https://github.com/DNature/nature-ui/issues"
  },
  "dependencies": {
    "@nature-ui/hooks": "^0.2.0",
    "@nature-ui/system": "^0.2.0",
    "@nature-ui/utils": "^0.2.0",
    "@nature-ui/visually-hidden": "^0.2.0",
    "@testing-library/jest-dom": "^5.11.6",
    "@testing-library/react": "^11.2.2",
    "@testing-library/react-hooks": "^3.4.2",
    "@testing-library/user-event": "^12.2.2",
    "@types/jest-axe": "^3.5.1",
    "jest-axe": "^4.1.0",
    "react-test-renderer": "^17.0.1"
  },
  "devDependencies": {
    "react": "^17.0.1"
  },
  "peerDependencies": {
    "react": "17.x"
  },
  "gitHead": "97ec99cb6e84c1feed8685a57529a1738d11a23d"
}
