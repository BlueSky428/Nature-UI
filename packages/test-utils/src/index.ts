export * from './test-utils';
export * from './press';
export * from '@testing-library/react';
export type {
  RenderHookOptions,
  RenderHookResult,
} from '@testing-library/react-hooks';
export * from 'jest-axe';
