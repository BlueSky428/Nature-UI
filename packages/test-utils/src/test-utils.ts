import '@testing-library/jest-dom/extend-expect';
import { fireEvent } from '@testing-library/react';
import { toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);

export { act as invoke, renderHook } from '@testing-library/react-hooks';

export { default as userEvent } from '@testing-library/user-event';
export const escape = (ui: HTMLElement) =>
  fireEvent.keyDown(ui, {
    key: 'Escape',
    keyCode: 27,
  });
