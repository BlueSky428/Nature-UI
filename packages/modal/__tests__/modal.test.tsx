import * as React from 'react';
import { render, axe, fireEvent, press } from '@nature-ui/test-utils';
import { PortalManager } from '@nature-ui/portal';

import {
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
} from '../src';

const renderWithPortal = (ui: React.ReactElement) =>
  render(<PortalManager>{ui}</PortalManager>);

describe('@nature-ui/modal', () => {
  test('should render correctly', () => {
    const tools = renderWithPortal(
      <Modal isOpen onClose={jest.fn()}>
        <ModalOverlay>
          <ModalContent>
            <ModalHeader>Modal header</ModalHeader>
            <ModalCloseButton />
            <ModalBody>Modal body</ModalBody>
            <ModalFooter>Modal footer</ModalFooter>
          </ModalContent>
        </ModalOverlay>
      </Modal>
    );

    expect(tools.asFragment()).toMatchSnapshot();
  });

  test('should have no accessibility violations', async () => {
    const tools = renderWithPortal(
      <Modal isOpen onClose={jest.fn()}>
        <ModalOverlay>
          <ModalContent>
            <ModalHeader>Modal header</ModalHeader>
            <ModalCloseButton />
            <ModalBody>Modal body</ModalBody>
            <ModalFooter>Modal footer</ModalFooter>
          </ModalContent>
        </ModalOverlay>
      </Modal>
    );

    const result = await axe(tools.container);

    expect(result).toHaveNoViolations();
  });

  test("should have the proper 'aria' attributes", () => {
    const tools = renderWithPortal(
      <Modal isOpen onClose={jest.fn()}>
        <ModalOverlay>
          <ModalContent data-testid='modal'>
            <ModalHeader>Modal header</ModalHeader>
            <ModalBody>Modal body</ModalBody>
          </ModalContent>
        </ModalOverlay>
      </Modal>
    );

    const dialog = tools.getByTestId('modal');

    /**
     * should have `aria-modal` set to `true`
     */
    expect(dialog).toHaveAttribute('aria-modal', 'true');
    expect(dialog).toHaveAttribute('role', 'dialog');

    /**
     * The id of `DialogBody` should equal the `aria-describedby` of the dialog
     */
    expect(tools.getByText('Modal body').id).toStrictEqual(
      dialog.getAttribute('aria-describedby')
    );

    /**
     * The id of `DialogHeader` should equal the `aria-labelledby` of the dialog
     */
    expect(tools.getByText('Modal header').id).toStrictEqual(
      dialog.getAttribute('aria-labelledby')
    );
  });

  test("should fire 'onClose' callback when close button is clicked", () => {
    const onClose = jest.fn();

    const tools = renderWithPortal(
      <Modal isOpen onClose={onClose}>
        <ModalOverlay>
          <ModalContent>
            <ModalHeader>Modal header</ModalHeader>
            <ModalCloseButton data-testid='close' />
          </ModalContent>
        </ModalOverlay>
      </Modal>
    );

    /**
     * click the close button
     */
    fireEvent.click(tools.getByTestId('close'));

    expect(onClose).toHaveBeenCalledWith(expect.anything());
  });

  test('clicking overlay or pressing "esc" calls the onClose callback', () => {
    const onClose = jest.fn();
    const tools = renderWithPortal(
      <Modal isOpen onClose={onClose}>
        <ModalOverlay data-testid='overlay'>
          <ModalContent>
            <ModalHeader>Modal header</ModalHeader>
            <ModalBody>Modal body</ModalBody>
          </ModalContent>
        </ModalOverlay>
      </Modal>
    );

    const overlay = tools.getByTestId('overlay');

    // FIXME: Get rid of the linting error when internet comes up.
    press.Escape(overlay);
    expect(onClose).toHaveBeenCalledWith();
  });
});
