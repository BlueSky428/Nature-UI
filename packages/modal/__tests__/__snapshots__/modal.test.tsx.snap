// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@nature-ui/modal should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="nature-portal"
    >
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="1"
      />
      <div
        data-focus-lock-disabled="false"
      >
        <div>
          <div
            class="flex justify-center fixed left-0 top-0 right-0 bottom-0 w-screen h-screen items-start z-40 overflow-auto css-514051-ModalOverlay"
            variant="blur"
          >
            <section
              aria-describedby="nature-modal--body-1"
              aria-labelledby="nature-modal--header-1"
              aria-modal="true"
              class="bg-white shadow-lg my-12 rounded flex flex-col relative focus:outline-none z-50 css-8v7m9r-ModalContent"
              id="nature-modal-1"
              role="dialog"
              tabindex="-1"
            >
              <header
                class="nature-modal__header p-4 font-bold text-xl css-1gfottg-StyledHeader"
                id="nature-modal--header-1"
              >
                Modal header
              </header>
              <button
                aria-label="Close"
                class="flex items-center justify-center flex-shrink-0 rounded focus:shadow-outline focus:outline-none transition duration-500 ease-in-out p-2 nature-modal__close-btn absolute top-0 right-0 mt-3 mr-3 css-ly3tul"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class=" css-9xkhqu-Icon"
                  focusable="false"
                  height="12"
                  role="presentation"
                  viewBox="0 0 24 24"
                  width="12"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <div
                class="nature-modal__body py-2 px-4 flex-1"
                id="nature-modal--body-1"
              >
                Modal body
              </div>
              <footer
                class="flex items-center justify-end p-4 css-w30ywn-ModalFooter"
              >
                Modal footer
              </footer>
            </section>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
    </div>
  </div>
</DocumentFragment>
`;
